src/main/scala/net/snowflake/spark/snowflake/SnowflakeJDBCWrapper.scala

```diff
@@ -556,7 +556,7 @@ private[snowflake] object DefaultJDBCWrapper extends JDBCWrapper {
       bindVariableEnabled: Boolean = true
     ): ResultSetMetaData =
       (ConstantString("select * from") + statement + "where 1 = 0")
-        .execute(bindVariableEnabled)(connection)
+        .prepareStatement(bindVariableEnabled)(connection)
```

Comment:
If we're just preparing the statement without executing it, we don't need the...`select ... where 1=0` wrapper anymore, right? We can just run `prepare` on the statement itself?

Reply:
There are some use cases that the pass-in parameter is table name only.
If we don't wrap it here, all the caller needs to wrap it.
So I think current fix is easier and low risk.